src/util/decoders.js

```diff
@@ -143,38 +144,54 @@ export async function constructDataFromBytes(bytes) {
       case '53': // Substrate UOS payload
         try {
           const crypto = firstByte === '00' ? 'ed25519' : firstByte === '01' ? 'sr25519' : null;
+          
           const pubKeyHex = uosAfterFrames.substr(6, 64)
           const publicKeyAsBytes = hexToU8a('0x' + pubKeyHex);
-          const ss58Encoded = encodeAddress(publicKeyAsBytes, 2); // encode to kusama
+          
           const hexEncodedData = '0x' + uosAfterFrames.slice(70);
           const rawPayload = hexToU8a(hexEncodedData);
+
           const isOversized = rawPayload.length > 256;
-  
           data['data']['crypto'] = crypto;
-          data['data']['account'] = ss58Encoded;
-  
+
+          let network;
+          let extrinsicPayload;
+
           switch (secondByte) {
             case '00': // sign mortal extrinsic
+              extrinsicPayload = new GenericExtrinsicPayload(rawPayload, { version: 3 });
+
               data['action'] = isOversized ? 'signData' : 'signTransaction';
               data['oversized'] = isOversized;
               data['isHash'] = isOversized;
               data['data']['data'] = isOversized
                 ? await blake2s(u8aToHex(rawPayload))
-                : new GenericExtrinsicPayload(rawPayload, { version: 3 });
+                : extrinsicPayload;
+
+              network = NETWORK_LIST[extrinsicPayload.genesisHash.toHex()];
+              data['data']['account'] = encodeAddress(publicKeyAsBytes, network.prefix); // encode to the prefix;
+
               break;
             case '01': // data is a hash
               data['action'] = 'signData';
               data['oversized'] = false;
               data['isHash'] = true;
               data['data']['data'] = rawPayload;
+              data['data']['account'] = encodeAddress(publicKeyAsBytes, 2); // default to Kusama
```

Comment:
can we use the prefix here as well rather than have it hardcoded directly.
This suggestion needs to import `SubstrateNetworkKeys` (also double check it, just typed it here)
```suggestion
              const kusamaNetworkKey = SubstrateNetworkKeys.KUSAMA;
              const kusamaPrefix = NETWORK_LIST[kusamaNetworkKey].prefix;
              data['data']['account'] = encodeAddress(publicKeyAsBytes, kusamaPrefix); // default to Kusama
```